packages/jest-did-matcher/src/matchers/toBeValidDid/predicate.js

```diff
@@ -1,4 +1,4 @@
 export default expected => {
-    const didRegex = /^did:(?<method>[a-z0-9]+):(?<idchar>([a-zA-Z0-9.\-_]|%[0-9a-fA-F]{2})+$)/;
+    const didRegex = /^did:(?<method>[a-z0-9]+)(:(?<idchar>([a-zA-Z0-9.\-_]|%[0-9a-fA-F]{2})+))+$/;
```

Comment:
Looks good, but this regex has no length upperlimit allowing it to be susceptible to a ReDoS attack. I'm concerned this regex will get copied from this test into an implementation. Should we be setting a max limit on the size of a DID?

Reply:
I have also realized that the ABNF allows for DIDs like did:example::::::abc:::123, while my regex does not, so it might need further tweaking.

A quick and dirty approach would be something like this (so it now matches 1:1 with the ABNF):
`^did:(?<method>[a-z0-9]+):((?<idchar>([a-zA-Z0-9.\-_]|%[0-9a-fA-F]{2})*):)*(?<idchar2>([a-zA-Z0-9.\-_]|%[0-9a-fA-F]{2})+)$`

But here I just copied & pasted the idchar-group, which is not ideal. Certainly one could come up with a cleaner regex, maybe by constructing it over multiple lines instead of cramping it into one line (or using some group-reusing inside the same regex, which would sacrifice readbility).

Your concerns regarding a ReDos attack seem valid, though I'm unsure of the implications on this (would one need to add this limit into the did-core spec itself? Or would it be okay to differ from the did-core spec there?)